sale_timesheet_hook/models/account_analytic_line.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2018-2019 Brainbean Apps
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, api, _
+from odoo.exceptions import ValidationError
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    @api.constrains('so_line', 'project_id', 'task_id')
+    def _check_sale_line_in_project_map(self):
+        """Since super method is not extendable, override it completely
+        with backwards compatibility"""
+        for timesheet in self:
```

Comment:
```suggestion
        for timesheet in self.filtered(lambda r: r.project_id or r.so_line) :
```

And we can remove the next if

Reply:
It's a bit different, but thanks for insight, updated!